### ISSUE TITLE
api: add a way to echo a highlighted message

### ISSUE THREAD

**Issue by notomo** [neovim/neovim#13672]
### Problem

I couldn't find the easy way to echo a highlighted message by api.
For example, implement `warn(msg)` is a bit annoying in lua.

```vim
function! s:warn(msg) abort
  echohl WarningMsg
  echomsg a:msg
  echohl None
endfunction
```

```lua
local warn = function(msg)
  vim.cmd("echohl WarningMsg")
  vim.cmd(([[echomsg "%s"]]):format(msg)) -- TODO: escape
  vim.cmd("echohl None")
end
```

In addition, `nvim_err_write()`, `nvim_out_write()` and `nvim_err_writeln()` don't accept highlight attributes.
(`echohl` doesn't work.)

### Solution (proposal)

Add `nvim_echo()` api.

```lua
local warn = function(msg)
  vim.api.nvim_echo(msg, {hl_group = "WarningMsg", history = true})
end
```

**Comment by cyansprite** [comment]
A workaround you could use so you only use one cmd instead of three ( change msg with format )

```lua
vim.cmd(([[echohl WarningMsg | echomsg "%s" | echohl None]]):format(msg))
```

:h :bar

**Comment by notomo** [comment]
Thanks!
But after all, `vim.cmd` requires to escape string (and split by newline).

```lua
local msg = "\"\\\na\n"
vim.cmd(([[echohl WarningMsg | echomsg "%s" | echohl None]]):format(vim.fn.escape(msg, "\"\\")))
```

**Comment by wookayin** [comment]
For reference: #13673 added the function whose signature is different than originally proposed here:

```
nvim_echo({chunks}, {history}, {opts})                           nvim_echo()  
                Echo a message.                                               
                                                                              
                Parameters:                                                   
                    {chunks}   A list of [text, hl_group] arrays, each        
                               representing a text chunk with specified       
                               highlight. hl_group element can be omitted     
                               for no highlight.                              
                    {history}  if true, add to message-history.         
                    {opts}     Optional parameters. Reserved for future use.
```

Example:
```
:lua vim.api.nvim_echo({{"A warning message: blahblah", "WarningMsg"}}, true, {})
```

Available in neovim 0.5.0+.  Personally speaking the API is not quite my taste as it seems less concise, but it works well :)
